File: problem-2/main.c
#include <stdio.h>
int fib(int n){
    if (n == 1 || n == 0){
        return 1;
    }
    else {
        return fib(n-1)+ fib(n-2);
    }
}
int main(){
    int sum = 0;
    int i = 1;
    while (1)
    {
        int x = fib(i);
        if (x > 4000000)
        {
            break;
        }
        if (x % 2 == 0) {
            sum += x;
        }
        i++;
    }
    printf("%d",sum);
    
}

File: problem-2/main.go
package main

func main(){
	var s uint64 = 0
	i := 1
	for {
		x := fib(uint64(i))
		if x > 4_000_000 {
			break
		} 
		if x % 2 == 0 {
			s += x
		}
		i++
	}
	println(s)
}
func fib(n uint64) uint64 {
	if n == 1 || n == 0 {
		return 1
	} else {
		return fib(n-1) + fib(n-2)
	}

}